Close #674: Rework escrow address usage

LeeSmet opened the pull request:
Generate 1 escrow address for a user, which is used to fund all
reservations for all farmers. We later split the payment, so that each
farmer gets the correct cut, in the explorer itself.

LeeSmet committed:
Close #674: Rework escrow address usage

Generate 1 escrow address for a user, which is used to fund all
reservations for all farmers. We later split the payment, so that each
farmer gets the correct cut, in the explorer itself.

zaibon commented on tools/explorer/pkg/escrow/escrow.go:
```diff
@@ -169,100 +173,109 @@ func (e *Escrow) checkReservations() error {
-			}
+		reservation, err := workloadtypes.ReservationFilter{}.WithID(escrowInfo.ReservationID).Get(e.ctx, e.db)
+		if err != nil {
+			slog.Error().Err(err).Msgf("failed to load reservation")
```
missing return ?

zaibon commented on tools/explorer/pkg/escrow/escrow.go:
```diff
@@ -169,100 +173,109 @@ func (e *Escrow) checkReservations() error {
-			}
+		pl, err := workloadtypes.NewPipeline(reservation)
+		if err != nil {
+			slog.Error().Err(err).Msgf("failed to process reservation in pipeline")
```
missing return ?

zaibon reviewed (changes requested):
found possible missing error return otherwise LGTM

LeeSmet committed:
Refactor the way we look for funded escrows

Rather than retrieve active escrows and loop over them in the same
function, isolate the code to check in a new function. This way we can
use actual error returns instead of error logs followed by continues,
and log just once in the caller.

zaibon commented on tools/explorer/pkg/escrow/escrow.go:
```diff
@@ -170,63 +169,89 @@ func (e *Escrow) checkReservations() error {
+			log.Error().
+				Str("address", escrowInfo.Address).
+				Int64("reservation_id", int64(escrowInfo.ReservationID)).
+				Err(err)
```
This line won't print anything zerolog requires to have `Msg`, `Msgf` or just `Send` if you don't want to add a message.
